src/transport/CryptoContext.cpp

```diff
@@ -72,17 +72,42 @@ CHIP_ERROR CryptoContext::InitFromSecret(const ByteSpan & secret, const ByteSpan
         infoLen = sizeof(RSEKeysInfo);
     }
 
+#if CHIP_CONFIG_SECURITY_TEST_MODE
+
+    // If enabled, override the generated session key with a known key pair
+    // to allow man-in-the-middle session key recovery for testing purposes:
+    //      44 D4 3C 91 D2 27 F3 BA 08 24 C5 D8 7C B8 1B 33
+    //      AC C1 8F 06 C7 BC 9B E8 24 6A 67 8C B1 F8 BA 3D
+
+    const char * kTestSecret    = "Test secret for key derivation";
+    size_t secretLen            = strlen(kTestSecret);
```

Comment:
The `kTestSecret` should be configurable, but set to this default, so that implementers can use their own known shared secret that may not be decodable by all.